pilot/pkg/proxy/envoy/v1/watcher.go

```diff
@@ -100,13 +105,21 @@ func (w *watcher) Run(ctx context.Context) {
 	<-ctx.Done()
 }
 
+// Reload tries to restart Envoy. However, if any of the required certs do not present, the reload will abort.
 func (w *watcher) Reload() {
 	config := BuildConfig(w.config, w.pilotSAN)
 
 	// compute hash of dependent certificates
 	h := sha256.New()
-	for _, cert := range w.certs {
-		generateCertHash(h, cert.Directory, cert.Files)
+	for _, cert := range w.requiredCerts {
```

Comment:
So what is the impact of this ? Can you add some comments ( including in the PR ) ? 

It seems envoy will not start - for how long ? How will it recover ? 
I'm not sure I understand how this will make things better - before the agent was watching the files, and when they showed up envoy was restarted. After the change - it's the same thing, except envoy doesn't run while the certs are missing ( which may save some CPU ).

Reply:
(Sorry for the misleading description, I have changed it.)
This PR is to force Envoy not start if auth is enabled and mtls certs are not in place -- yes, I think the only thing currently this PR does is to save the CPU. But I think this is the right thing to fix...
In the future, with the health check ready, I think the caller can identify the service is not ready and do retries based on this (different from the retries due to network error).

I could drop this PR if you think the benefit is not obvious and it introduces complexity. But I doubt in the future we may want this.